Examples/Io/Root/src/RootTrackStatesWriter.cpp

```diff
@@ -332,16 +333,27 @@ ActsExamples::ProcessCode ActsExamples::RootTrackStatesWriter::writeT(
     }
 
     // Get the trackStates on the trajectory
-    m_nParams = {0, 0, 0};
+    m_nParams = {0, 0, 0, 0};
+
     for (const auto& state : track.trackStatesReversed()) {
-      // we only fill the track states with non-outlier measurement
-      auto typeFlags = state.typeFlags();
-      if (!typeFlags.test(Acts::TrackStateFlag::MeasurementFlag)) {
+      const auto& surface = state.referenceSurface();
+
+      // get the geometry ID
+      auto geoID = surface.geometryId();
+      m_volumeID.push_back(geoID.volume());
```

Comment:
Won't that result in vectores of different lengths if any state has no uncalibrated source-link? Or do I oversee something here?